core/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -648,6 +657,144 @@ public boolean equals(Object other) {
     }
   }
 
+  /**
+   * Arguments for {@link Helper#createSubchannel(CreateSubchannelArgs)}.
+   *
+   * @since 1.21.0
+   */
+  public static final class CreateSubchannelArgs {
+    private final List<EquivalentAddressGroup> addrs;
+    private final Attributes attrs;
+    private final SubchannelStateListener stateListener;
+
+    private CreateSubchannelArgs(
+        List<EquivalentAddressGroup> addrs, Attributes attrs,
+        SubchannelStateListener stateListener) {
+      this.addrs = checkNotNull(addrs, "addresses are not set");
+      this.attrs = checkNotNull(attrs, "attrs");
+      this.stateListener = checkNotNull(stateListener, "SubchannelStateListener is not set");
+    }
+
+    /**
+     * Returns the addresses, which is an unmodifiable list.
+     */
+    public List<EquivalentAddressGroup> getAddresses() {
+      return addrs;
+    }
+
+    /**
+     * Returns the attributes.
+     */
+    public Attributes getAttributes() {
+      return attrs;
+    }
+
+    /**
+     * Returns the state listener.
+     */
+    public SubchannelStateListener getStateListener() {
+      return stateListener;
+    }
+
+    /**
+     * Returns a builder with the same initial values as this object.
+     */
+    public Builder toBuilder() {
+      return newBuilder().setAddresses(addrs).setAttributes(attrs).setStateListener(stateListener);
+    }
+
+    /**
+     * Creates a new builder.
+     */
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public static final class Builder {
+      private List<EquivalentAddressGroup> addrs;
+      private Attributes attrs = Attributes.EMPTY;
+      private SubchannelStateListener stateListener;
+
+      Builder() {
+      }
+
+      /**
+       * The addresses to connect to.  All addresses are considered equivalent and will be tried
+       * in the order they are provided.
+       */
+      public Builder setAddresses(EquivalentAddressGroup addrs) {
+        this.addrs = Collections.singletonList(addrs);
+        return this;
+      }
+
+      /**
+       * Required.  The addresses to connect to.  All addresses are considered equivalent and will
```

Comment:
Having "Required." as its own sentence here hurts the readability of the javadoc. Javadoc using the first sentence as its summary, so basically this won't have any documentation in the summary view at the top of the page.